crates/noirc_evaluator/src/ssa_refactor/ir/map.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::HashMap;
+
+/// A unique ID corresponding to a value of type T.
+/// This type can be used to retrieve a value of type T from
+/// either a DenseMap<T> or SparseMap<T>.
+///
+/// Note that there is nothing in an Id binding it to a particular
```

Comment:
Yeah, this is a good point -- If we do end up in a situation where we have DenseMap<T> and SparseMap<T> both being used at the same time in a lot of places, it will make sense to revise these structures